src/product_config_utils.rs

```diff
@@ -9,12 +9,28 @@ use tracing::{debug, error, warn};
 pub enum ConfigError {
     #[error("Invalid configuration found")]
     InvalidConfiguration,
+
+    #[error("Collected product config validation errors: {collected_errors:?}")]
+    ProductConfigErrors {
+        collected_errors: Vec<product_config::error::Error>,
+    },
 }
 
+/// This needs to be implemented by every generic (T) struct that may appear in [Role<T>]
+/// or the top level in order to determine where config properties are configured.
+///
+/// The options are:
+/// - Environmental variables (env)
+/// - Command line arguments (cli)
+/// - Configuration files (files)
+///
+/// Returned empty Maps will be ignored. Check out `to_hash_map(...)` in
+/// https://github.com/stackabletech/product-config/blob/main/src/ser.rs
+/// if you do not need to differentiate between the options and want to serialize all properties
+/// in a certain config T to one option.
```

Comment:
```suggestion
/// Returned empty Maps will be ignored.
/// Check out [`crate::ser::to_hash_map`] if you do not need to convert a struct to a HashMap in an easy way..
```

Reply:
I think we lose info here. And i would use to_hash_map if i want an **easy** way to convert and put everything e.g. in env variables.

Reply:
You definitely want to fix the rust doc link. And yes, my comment was wrong. It should be `if you do need`.

That said: I don't understand the original sentence: `if you do not need to differentiate between the options and want to serialize all properties in a certain config T to one option.`

@soenkeliebau  as someone who hasn't been part of this. Could you say if this comment works for you as is?

Reply:
I don't really understand the initial comment I'm afraid.